cloudflare/index.js

```diff
@@ -0,0 +1,354 @@
+  }
+  catch (err)
+  {
+    console.error(err)
+    return new Response(err.stack, { status: 500, headers:corsHeaders })
+  }
+}
+
+async function handleZoomRegistrationRequest(request) {
+  try
+  {
+    const formData = await request.formData()
+    const bodydata = {}
+    for (const entry of formData.entries()) {
+      bodydata[entry[0]] = entry[1]
+    }
+
+    // Validate the submitted data
+    if (!validateRegistrationFormData(bodydata)) {
+      return new Response('Invalid submission', { status: 400, headers:corsHeaders })
+    }
+
+    // Extract the recaptcha token
+    const recaptchaToken = bodydata['g-recaptcha-response']
+    if (!recaptchaToken) {
+      return new Response('Invalid reCAPTCHA', { status: 400, headers:corsHeaders })
+    }
+
+    const recaptchaResponse = await fetch(
+      `https://www.google.com/recaptcha/api/siteverify?secret=${RECAPTCHASECRET}&response=${recaptchaToken}`, {
+        method: 'POST'
+    })
+
+    const recaptchaBody = await recaptchaResponse.json()
+    if (recaptchaBody.success == false) {
+      return new Response('reCAPTCHA failed', { status: 400, headers:corsHeaders })
+    }
+
+    // At this point, we passed the captcha and we have valid entries
+
+    const payload = {}
+    payload["first_name"] = bodydata["firstname"]
+    payload["last_name"] = bodydata["lastname"]
+    payload["email"] = bodydata["address"]
+    payload["org"] = bodydata["affiliation"]
+    payload["custom_questions"] = [{
+      "title": "Please confirm you agree to follow the participant instructions: http://virtualscienceforum.org/#/attendeeguide",
+      "value": "Yes",
+      }]
+    payload["auto_approve"] = 1
+
+    var jwt = require('jsonwebtoken');
+    var token = jwt.sign({ iss: ZOOMAPIKEY }, ZOOMAPISECRET, { algorithm: 'HS256', expiresIn: '1h' });
+
+    let requestbody = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded",
+        "Content-Length": payload.length,
+        'Authorization': 'Bearer {' + token + '}'
+      },
+      body: urlfy(payload),
+    }
+
+    // Send to Zoom
+    const meetingId = bodydata['meetingID']
+    var registerURL = 'https://api.zoom.us/v2/meetings/' + meetingId + '/registrants'
+    const response = await fetch(registerURL, requestbody)
+
+    // If we get here, we managed to register
+
+    // Add to mailing list!
+    const username = bodydata["firstname"] + " " + bodydata["lastname"]
+    const useremail = bodydata["address"]
+    const eventtype = bodydate["eventType"]
+
+    var mailgunListName = "";
+    switch( eventtype ) {
+      case "lrc":
+        mailgunListName = "vsf-announce"
+        break;
+      case "speakers_corner":
+        mailgunListName = "speakers_corner"
+        break;
+      default:
+        return new Response("The list " + eventtype + " cannot be subscribed to.", {status:400, headers:corsHeaders})
```

Comment:
Returning over here would result in an error message even though the registration goes through. We should either append to the result or let this slide.

Reply:
Do you have a preference?

Reply:
Adding to the response message "Unable to sign up to the mailing list, please report the problem to the VSF." is the best.